### ISSUE TITLE
[Server-side] Support additional filtering options for the twingate_group and twingate_groups data sources

### ISSUE THREAD

**Issue by alexmensch** [Twingate/terraform-provider-twingate#138]
With the implementation of #137 and the future support of additional filtering options in the Twingate Admin API, we'll be able to support additional filtering options on the `twingate_group` data source. We will further define the implementation later on, but it would be reasonable to support filtering on any attribute that is exported by the data source.

**Comment by tjstansell** [comment]
Something to think about with regard to this.  You could introduce your own level of filtering within this data source that has nothing to do with the API filtering capabilities.  Obviously, as the API is updated and you're able to optimize your calls to only return the data you care about, that's beneficial.  I wouldn't think it would be a requirement, however.  This is functionally the same as the end user having to do their own filtering, just cleaner/nicer.
```hcl
data "twingate_groups" "okta" {
  isActive = true
  type     = "SYNCED"
}
```
is a lot nicer to write than
```hcl
data "twingate_groups" "all" {}
locals {
   okta_groups = [ for i in data.twingate_groups.all.groups : i if i.type == "SYNCED" && i.isActive == true ]
}
```

**Comment by alexmensch** [comment]
Update: we've decided that we'll prioritize server-side filtering in our public API rather than implement filtering in the provider, so I've closed #141 in reflection of this decision.
